openassessment/xblock/grade_mixin.py

```diff
@@ -584,3 +590,101 @@ def _assessment_grade_context(self, assessment):
                     part['option']['label'] = option_labels.get(option_label_key, part['option']['name'])
 
         return assessment
+
+    def _get_assessment_type(self, workflow):
+        # assessment_step would always have staff in it, so be careful while
+        # making decisions upon it
+        _assessment_type = None
+        score = workflow['score']
+        complete = score is not None
+
+        is_self_assessment = "self-assessment" in self.assessment_steps
+        is_peer_assessment = "peer-assessment" in self.assessment_steps
+        is_staff_assessment = "staff-assessment" in self.assessment_steps
+
+        if is_staff_assessment:
+            _assessment_type = "staff_defined"
+        else:
```

Comment:
Simplify the code below by returning early and avoiding more if-else statements.